agent/src/main/java/org/openremote/agent/protocol/http/HTTPProtocol.java

```diff
@@ -346,7 +377,7 @@ protected void doStart(Container container) throws Exception {
         webTargetBuilder.followRedirects(followRedirects);
 
         LOG.fine("Creating web target client '" + baseUri + "'");
-        webTarget = webTargetBuilder.build();
+        webTarget = webTargetBuilder.build(new DynamicTimeInjectionFilter(timerService::getCurrentTimeMillis), new QueryParameterInjectorFilter());
```

Comment:
Because the WebTargetBuilder instantiates the filters by itself and applies them to the target. Thing is, the WebTargetTest fails if we don't supply both filters. The DynamicTimeFilter requires a currentMillisSupplier, so just  giving it the static reference doesn't do. Not all WebTargetBuilders require the filters, so injecting them where needed is a good way. What I can do though is at least use a static reference to the queryParameterInjectorFilter, and then inject just the dynamic time filter.

Reply:
This comment still needs answering

Reply:
`DynamicTimeFilter` could use a no arg constructor and then inject the time into the request context properties like the dynamic value for the `QueryParameterInjectorFilter` only if the property is set do you then search for `{$time}` strings to replace